drake/core/test/vector_test.cc

```diff
@@ -0,0 +1,143 @@
+#include "drake/examples/Pendulum/Pendulum.h"  // to get some types
+#include "drake/util/testUtil.h"
+
+#include "drake/util/eigen_matrix_compare.h"
+#include "gtest/gtest.h"
+
+using Drake::CombinedVector;
+using Drake::size;
+using Drake::CombinedVectorUtil;
+using Drake::NullVector;
+using Drake::InputOutputRelation;
+using drake::util::MatrixCompareType;
+
+namespace drake {
+namespace test {
+
+// Tests the ability to set a PendulumState equal to a vector and vice versa.
+TEST(VectorTest, ValueAssignment) {
+  Eigen::Vector2d x;
+  x << 0.2, 0.4;
+
+  PendulumState<double> state;
+  state.theta = 0.2;
+  state.thetadot = .3;
+
+  EXPECT_EQ(size(state), static_cast<size_t>(2));
```

Comment:
I removed the static cast and this unit test continued to work on both Ubuntu Linux and OS X. I haven't finished spinning up my Windows development platform but will assume the static cast is not necessary and rely on the CI servers to detect and invalidate this assumption.